rust-runtime/aws-smithy-runtime-api/src/client/http/request.rs

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::box_error::BoxError;
+use crate::client::interceptors::context::Error;
+use aws_smithy_http::body::SdkBody;
+use http as http0;
+use std::borrow::Cow;
+use std::fmt::{Debug, Formatter};
+use std::str::FromStr;
+
+/// An HTTP Request Type
+pub struct Request<B = SdkBody> {
+    inner: http0::Request<B>,
+}
+
+impl<B: Debug> Debug for Request<B> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.inner)
+    }
+}
+
+impl<B> Request<B> {
+    /// Converts this request into an http 0.x request.
+    ///
+    /// Depending on the internal storage type, this operation may be free or it may have an internal
+    /// cost.
+    pub fn into_http0x(self) -> http::Request<B> {
+        self.inner
+    }
+
+    /// Returns a reference to the header map
+    pub fn headers(&self) -> Headers<'_> {
+        Headers::from(&self.inner)
+    }
+
+    /// Returns a mutable reference to the header map
+    pub fn headers_mut(&mut self) -> HeadersMut<'_> {
+        HeadersMut::from(&mut self.inner)
+    }
+
+    /// Returns the body associated with the request
+    pub fn body(&self) -> &B {
```

Comment:
Would need `body_mut` too.

Also, it would be super nice to have a conveniences like `map_body`/`take_body` since we frequently have to juggle `mem::swap`s in the orchestrator code to accomplish that.

Reply:
yeah there are a bunch of methods that need to be added to support everything